src/compiler/types.ts

```diff
@@ -422,6 +422,7 @@ namespace ts {
         JavaScriptFile =     1 << 20, // If node was parsed in a JavaScript
         ThisNodeOrAnySubNodesHasError = 1 << 21, // If this node or any of its children had an error
         HasAggregatedChildData = 1 << 22, // If we've computed data from children and cached it in this node
+        InJSDocNamespace = 1 << 23, // if the node is a member in a JSDoc namespace

```

Comment:
I think this is only used on Identifiers, so it might be better to not use up a NodeFlag and instead put an optional property on Identifiers. I think the code that uses that would only a be a bit more complex.